compiler/qsc/src/interpret/stateless.rs

```diff
@@ -44,12 +44,35 @@ enum ErrorKind {
     Eval(#[from] qsc_eval::Error),
 }
 
+pub struct Interpreter {
+    context: Context,
+}
+
+#[self_referencing]
 pub struct Context {
     store: PackageStore,
+    env: Env,
+    sim: SparseSim,
+    #[borrows(store)]
+    #[covariant()]
+    lookup: Lookup<'this>,
     package: PackageId,
+    #[borrows(package)]
+    #[not_covariant()]
+    state: State<'this>,
+}
```

Comment:
I think we should try to avoid ouroboros... It should be possible to split `Context` into two structs, where the struct containing `Lookup` and `State` has a lifetime tied to the struct containing `PackageStore`, right?